src/Microsoft.ML.OnnxConverter/SaveOnnxCommand.cs

```diff
@@ -322,6 +352,30 @@ private void Run(IChannel ch)
                     nameof(Arguments.LoadPredictor), "We were explicitly told to load the predictor but one was not present.");
             }
 
+            // Convert back to values the KeyDataViewType "pass-through" columns
+            // (i.e those that remained untouched by the model). This is done to enable NimbusML to get these values
+            // as described in https://github.com/dotnet/machinelearning/pull/4841
+
+            var inputKeyDataViewTypeColumnsNames = new HashSet<string>();
+            foreach (var col in source.Schema)
+                if (col.IsHidden == false && HasKeyValues(col))
+                    inputKeyDataViewTypeColumnsNames.Add(col.Name);
+
+            var passThroughColumnNames = new HashSet<string>();
+            var onlyDistinctColumnNames = end.Schema.Select(c => c.Name).Distinct(); //  only check column names that appear once in the output schema
+            foreach (var col in end.Schema)
+                if (col.IsHidden == false && onlyDistinctColumnNames.Contains(col.Name) && HasKeyValues(col))
```

Comment:
>onlyDistinctColumnNames [](start = 45, length = 23)

This will contain all the columns in the output schema, just without duplicates.

Reply:
I think something like this might work:
```
foreach (var col in end.Schema)
    if (inputKeyDataViewTypeColumnNames.Contains(col.Name) && col.IsHidden)
        inputKeyDataViewTypeColumnNames.Remove(col.Name));
```

(I don't know if `HashSet` has a `Remove` method, perhaps you'll need a different kind of collection?)

---
In reply to: [383573577](https://github.com/dotnet/machinelearning/pull/4841#discussion_r383573577) [](ancestors = 383573577)

Reply:
You're right. I'll fix it.